pkg/apis/secretgen2/v1alpha1/secret_export.go

```diff
@@ -68,15 +94,28 @@ func (e SecretExport) Validate() error {
 	var errs []error
 
 	toNses := e.StaticToNamespaces()
+	toSmf := e.Spec.ToSelectorMatchFields
 
-	if len(toNses) == 0 {
-		errs = append(errs, fmt.Errorf("Expected to have at least one non-empty to namespace"))
+	if len(toNses) == 0 && len(toSmf) == 0 {
+		errs = append(errs, fmt.Errorf("Expected to have at least one non-empty to namespace or to namespace annotation"))
 	}
 	for _, ns := range toNses {
 		if len(ns) == 0 {
 			errs = append(errs, fmt.Errorf("Expected to namespace to be non-empty"))
 		}
 	}
+	for _, s := range toSmf {
```

Comment:
Is the behaviour of this PR intended to be that your SecretExport needs BOTH `toNamespaces` and the selector to match? My understanding of the issue raised was that you wanted to use the annotation only, and no need for `toNamespaces` at all. Right?

```
---
apiVersion: secretgen.carvel.dev/v1alpha1
kind: SecretExport
metadata:
  name: secret
  namespace: sg-test1
spec:
  toNamespaces:
  - sg-test2
  - sg-test3
  toSelectorMatchFields:
  - key: "metadata.annotations.field\\.cattle\\.io/projectId"
    operator: In
    values:
    - "cluster1:project1"
```

Reply:
not both are needed, but they can be combinated, for the simplicity of testing I added the feature implem to an existing secret that is also exporting to ns (not the same as the ns matched by `toSelectorMatchFields`), but if you prefer I can implement it on another new secret